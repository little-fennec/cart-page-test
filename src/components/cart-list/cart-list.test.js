import React from 'react';
import CartList from './index';
import Enzyme, {shallow, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import StoreService from '../../services/store-service';
import StoreServiceContext from '../../components/store-service-context';
import configureMockStore from 'redux-mock-store';

const storeService = new StoreService();
const mockStore = configureMockStore();

Enzyme.configure({ adapter: new Adapter() });

describe('Screen <CartList/>', () => {

    const initialState = {
        loading: false,
        cartItems: []
    };

    const store = mockStore(initialState);

    it('renders correctly ', () => {
        const wrapper = mount(
            <Provider store={store}>
                <StoreServiceContext.Provider value={storeService}>
                    <CartList/>
                </StoreServiceContext.Provider>
            </Provider>
        );
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });

});
