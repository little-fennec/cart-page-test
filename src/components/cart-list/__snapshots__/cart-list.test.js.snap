// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen <CartList/> renders correctly  1`] = `
<div
  className="cart-list"
>
  <div
    className="cart-list__clear-all"
    onClick={[Function]}
  >
    Clear all
  </div>
  <div
    className="cart-summary"
  >
    <div
      className="cart-summary__title"
    >
      Cart Summary
    </div>
    <div
      className="cart-summary__row"
    >
      <span>
        Total items
      </span>
      <span />
    </div>
    <div
      className="cart-summary__row free"
    >
      <span>
        Free delivery
      </span>
      <span>
        0$
      </span>
    </div>
    <div
      className="cart-summary__row"
    >
      <span>
        Total price
      </span>
      <span>
        $
      </span>
    </div>
    <div
      className="btn"
    >
      Proceed to checkout
    </div>
  </div>
</div>
`;

exports[`Screen/ServiceListingScreen renders correctly  1`] = `
<div
  className="cart-list"
>
  <div
    className="cart-list__clear-all"
    onClick={[Function]}
  >
    Clear all
  </div>
  <div
    className="cart-summary"
  >
    <div
      className="cart-summary__title"
    >
      Cart Summary
    </div>
    <div
      className="cart-summary__row"
    >
      <span>
        Total items
      </span>
      <span />
    </div>
    <div
      className="cart-summary__row free"
    >
      <span>
        Free delivery
      </span>
      <span>
        0$
      </span>
    </div>
    <div
      className="cart-summary__row"
    >
      <span>
        Total price
      </span>
      <span>
        $
      </span>
    </div>
    <div
      className="btn"
    >
      Proceed to checkout
    </div>
  </div>
</div>
`;
